mlflow/metrics/metric_definitions.py

```diff
@@ -363,3 +358,29 @@ def _fn(predictions, targets):
         return MetricValue(scores=scores, aggregate_results=standard_aggregations(scores))
 
     return _fn
+
+
+def _recall_at_k_eval_fn(k):
+    def _fn(predictions, targets):
+        if not _validate_list_str_data(
+            predictions, "precision_at_k", predictions_col_specifier
+        ) or not _validate_list_str_data(targets, "precision_at_k", targets_col_specifier):
+            return
+
+        scores = []
+        for i in range(len(predictions)):
+            # only include the top k retrieved chunks
+            ground_truth, retrieved = set(targets[i]), set(predictions[i][:k])
```

Comment:
/ai rewrite this using zip (assuming predictions and targets are the same size). No explanation is required. Just give me code.

Reply:
predictions and targets are the same size, so this works! thanks for the recommendation